aten/src/ATen/InferSize.h

```diff
@@ -19,7 +19,7 @@ namespace at {
 template <typename InputArrayRef, typename NumelType, typename ResultVec>
 inline void infer_size_impl(
     InputArrayRef shape,
-    NumelType numel,
+    const NumelType& numel,
```

Comment:
This is not unconditionally an improvement. When NumelType is an int64_t you have introduced an unnecessary reference here. In fact, SymInt also is a single int64_t size struct, and although C++ ABI shenanigans prevent it from being passed in a register, it definitely could be!

Reply:
Make sense, reverted